File: strings/151-reverse-words-in-a-string/solution.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
  let arr = s.split(/\s+/);

  let result = [];

  for(let x=arr.length-1; x>=0; x--){
   
    if(arr[x] !== ''){
      result.push(arr[x]);
    }

  }
    
  return result.join(' ');
};
console.log(reverseWords('the sky is blue'));
console.log(reverseWords('  hello world  '));
console.log(reverseWords('a good   example'));

File: strings/151-reverse-words-in-a-string/solution.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s = s.split()
        s = s[::-1]
        return ' '.join(s)


# Add this block to run the code locally in VS Code
if __name__ == "__main__":
    solver = Solution()
    # Replace with the test cases from the LeetCode problem
    result = solver.reverseWords('the sky is blue')
    print(f"Output: {result}")

    result = solver.reverseWords('a good   example')
    print(f"Output: {result}")

    result = solver.reverseWords('  hello world  ')
    print(f"Output: {result}")